client/hotrod-client/src/test/java/org/infinispan/client/hotrod/HotRodIntegrationTest.java

```diff
@@ -351,6 +355,52 @@ public void testPutSerializableByteArray() {
       assertArrayEquals(bytes, binaryRemote.get("aKey"));
    }
 
+   public void testComputeConcurrently() throws Exception {
+      String key = "key";
+      String value = "value";
+      String anotherValue = "anotherValue";
+      CheckPoint checkPoint = new CheckPoint();
+
+      // Execute compute async. This notifies the compute mapping is executing and blocks.
+      // We keep track of how many times the callback executes, since it needs to retry because of the concurrent creation.
+      AtomicInteger retries = new AtomicInteger();
+      CompletableFuture<String> first = remoteCache.computeAsync(key, (k, v) -> {
+         if (retries.incrementAndGet() == 1) {
+            assertThat(k).isEqualTo(key);
+            assertThat(v).isNull();
+            checkPoint.trigger("computed_first");
+            try {
+               checkPoint.awaitStrict("computed_first_proceed", 10, TimeUnit.SECONDS);
+            } catch (InterruptedException | TimeoutException e) {
+               throw new RuntimeException("Failed receiving event", e);
+            }
+         }
+         return value;
+      });
+
```

Comment:
We need to add a `checkpoint.awaitStrict("computed_first", 10, TimeUnit.SECONDS` here to make sure the other remote call is inside the compute block.

Reply:
Good catch!